framework/device/cpu/topology.cpp

```diff
@@ -1562,6 +1562,97 @@ void restrict_topology(DeviceRange range)
         topo = build_topology();
 }
 
+void analyze_test_failures_for_topology(const struct test *test,
+        const PerThreadFailures &per_thread_failures, bool all_threads_failed_equally)
+{
+    Topology topology  = Topology::topology();
+    if (!topology.isValid()) {
+        // can't use this information
+        if (all_threads_failed_equally)
```

Comment:
`all_threads_failed_equally = true` is impossible, because the caller never reaches the point of calling us with that value. Don't pass it to this function and remove the impossible conditions.

We redefine a variable with the same name below.